staging/src/k8s.io/kubectl/pkg/describe/versioned/describe.go

```diff
@@ -3913,6 +3915,61 @@ func describeStorageClass(sc *storagev1.StorageClass, events *corev1.EventList)
 	})
 }
 
+type CSINodeDescriber struct {
+	clientset.Interface
+}
+
+func (c *CSINodeDescriber) Describe(namespace, name string, describerSettings describe.DescriberSettings) (string, error) {
+	csi, err := c.StorageV1().CSINodes().Get(name, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	var events *corev1.EventList
+	if describerSettings.ShowEvents {
+		events, _ = c.CoreV1().Events(namespace).Search(scheme.Scheme, csi)
+	}
+
+	return describeCSINode(csi, events)
+}
+
+func describeCSINode(csi *storagev1.CSINode, events *corev1.EventList) (output string, err error) {
+	metadata := csi.GetObjectMeta()
+	return tabbedString(func(out io.Writer) error {
+		w := NewPrefixWriter(out)
+		w.Write(LEVEL_0, "Name:\t%s\n", csi.GetName())
+		printLabelsMultiline(w, "Labels", csi.GetLabels())
+		printAnnotationsMultiline(w, "Annotations", csi.GetAnnotations())
+		w.Write(LEVEL_0, "Metadata:\n")
+		w.Write(LEVEL_1, "Creation Timestamp:\t%s\n", csi.CreationTimestamp.Time.Format(time.RFC1123Z))
+		for _, owner := range metadata.GetOwnerReferences() {
+			w.Write(LEVEL_1, "Owner References:\n")
+			w.Write(LEVEL_2, "API Version:\t%s\n", owner.APIVersion)
+			w.Write(LEVEL_2, "Kind:\t%s\n", owner.Kind)
+			w.Write(LEVEL_2, "Name:\t%s\n", owner.Name)
+			w.Write(LEVEL_2, "UID:\t%s\n", owner.UID)
+		}
+		w.Write(LEVEL_1, "Resource Version:\t%s\n", metadata.GetResourceVersion())
+		w.Write(LEVEL_1, "Self Link:\t%s\n", metadata.GetSelfLink())
+		w.Write(LEVEL_1, "UID:\t%s\n", metadata.GetUID())
```

Comment:
I don't think it's common to show this metadata in describe.

Reply:
I've removed the metadata here.